### PR TITLE
solaris based systems update

### PR DESCRIPTION
**Pull request by devnexen**
basis to do cpu affinity with some illumos specifics.

### CODE REVIEW

File: src/unix/solarish/mod.rs [hunk 1]
```diff
@@ -2227,6 +2241,8 @@ extern "C" {
     pub fn labs(i: ::c_long) -> ::c_long;
     pub fn rand() -> ::c_int;
     pub fn srand(seed: ::c_uint);
+    pub fn getentropy(buf: *mut ::c_void, buflen: ::size_t) -> ::c_int;
+    pub fn getrandom(bbuf: *mut ::c_void, buflen: ::size_t, flags: ::c_uint);
```
Comment (JohnTitor) [inline src/unix/solarish/mod.rs:2245]: Shouldn't it return `ssize_t`?

Comment (devnexen) [reply src/unix/solarish/mod.rs:2245]: oh right I just forgot the return type

Comment (RalfJung) [reply src/unix/solarish/mod.rs]: Is using the same type for Illumos and Solaris correct here? `rand`/`getrandom` used to use different return types for these two OSes (see e.g. https://github.com/rust-random/rand/pull/730), and glibc also mentions Solaris having a non-standard return type ([here](https://www.gnu.org/software/gnulib/manual/html_node/getrandom.html)).

**Pull request merged**